### PR TITLE
Base+Documentation: Add UsingFontEditor markdown file

### PR DESCRIPTION
**Pull request by djwisdom**
My initial attempt at documenting FontEditor for those so inclined
to trying out creating fonts via FontEditor on SerenityOS.

### CODE REVIEW

File: Documentation/FontEditor/UsingFontEditor.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+![](FontEditor.png)
+
+### Basic Parts
+FontEditor has the following basic parts:
+1. The menu bar at the top
+2. The toolbar
+3. The main workspace
+4. The status bar at the bottom
+
+Presently, you can resize the entire FontEditor form to your liking. You can turn-on or turn-off the visibility of Font Metadata and Unicode Blocks. You can change the glyph editor window zoom factor.  All of these things are currently available in FontEditor automatically.
```
Comment (EWouters) [inline Documentation/FontEditor/UsingFontEditor.md:14]: You can turn the visibility of Font Metadata and Unicode Blocks on or off.

File: Documentation/FontEditor/UsingFontEditor.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
...) empty or gray. On state is displayed as black by default. Off state is displayed as white by default. Empty state is displayed as gray by default. Off state or white is relevant most specially if the font is fixed-width as it dictates the distance a glyph will have from left, center or right. Empty state or gray is important for variable-width font as it affects the overall width of the glyph. For variable width fonts, you move the glyph flushed left on the glyph editor window and remove any extra empty grids by decrementing the present column counter located right under the glyph editor window and thereby leaving only the entire glyph all by itself. Below the glyph editor is the glyph tool which contains the following: pen icon for creating glyph itself, move icon to enable the entire glyph move to top, down, left or right from within the glyph editor window. Below the glyph tool are the transform icons which are flip horizontal, flip vertical, rotate counter-clockwise 90degress and rotate clockwise 90degrees. Transform tools are most helpful when copying over existing glyphs and transforming them to form a new glyph. A becomes V, M becomes W, c becomes e, n becomes u, etcetera.
```
Comment (EWouters) [inline Documentation/FontEditor/UsingFontEditor.md:52]: Zoom scale -> zoom level
This is where you will spend about 95% of the time
90degress -> 90°

File: Documentation/FontEditor/UsingFontEditor.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+- The ***right section*** which displays both search input for unicode blocks and the unicode block selection list.
+Metadata and unicode blocks can be turned on or off via ` Menu -> View ` option. Unicode block filters only what to display contained within the range of the block. Basic Latin covers 000-007F, Latin-1 Supplement covers 0080-00FF, Latin Extended-A covers 0100-017F and so on and so forth. Global search for glyph is affected when a certain block is currently selected. Only by selecting ` Show All ` will the global glyph search work as expected. So make it a habit of confirming that Show All is active before searching for a glyph.
```
Comment (EWouters) [inline Documentation/FontEditor/UsingFontEditor.md:57]: Global search for *a* glyph

File: Documentation/FontEditor/UsingFontEditor.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+### The status bar
+The status bar displays additional information describing what each menu entry and toolbar icon does. It identifies what is the current unicode value of the glyph the cursor is in. It shows the glyph's visual representation (if available), glyph description and glyph dimensions. Second only to the glyph editor window, the status bar is your next best friend on your path to becoming a font master.
```
Comment (EWouters) [inline Documentation/FontEditor/UsingFontEditor.md:60]: It identifies the unicode value of the glyph currently under the cursor.
glyph description -> description
glyph dimensions -> dimensions

File: Documentation/FontEditor/UsingFontEditor.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+### Create New Font
+![](FontEditor_New_Font.png)
+
+To create a new font, you can either click on New Font icon on the Toolbar or go to ` File -> New Font ` on the Menubar.
+A wizard will walk you into preparing the needed parameters for your new font.
```
Comment (EWouters) [inline Documentation/FontEditor/UsingFontEditor.md:66]: into -> through

File: Documentation/FontEditor/UsingFontEditor.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+### Typeface Properties
+![](FontEditor_Typeface_properties.png)
+
+You can try out the default values just to get the feel of the program.
+
+### Glyph Properties
+![](FontEditor_Glyph_properties.png)
+
+Just click **Finish** if you are so inclined.
```
Comment (EWouters) [inline Documentation/FontEditor/UsingFontEditor.md:76]: if -> when

File: Documentation/FontEditor/UsingFontEditor.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+### Edit Glyph Properties
+![](FontEditor_Edit_Glyph_properties.png)
+
+Adjust values to suit your needs. The higher the value, the larger the font size.
```
Comment (EWouters) [inline Documentation/FontEditor/UsingFontEditor.md:81]: Adjust the values

File: Documentation/FontEditor/UsingFontEditor.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+Congratulations on your successful initial font setup. Now you are ready to begin. And begin you shall. There is more to font creation than simply scribbling away. You need to always remind yourself this question: "Where is the fun in that?". Start with what you need to do while striving to achieve fun in the process. Let's be honest, font creation is one of the many thankless jobs, unless and until there is fun in it, why bother?
+
+### Launch another instance
+![](FontEditor_Launching_second_instance.png)
+
+Figure above shows where you can find FontEditor from inside the SerenityOS desktop. It is joined together with other similarly amazing applications like GML Playground, Hack Studio, Hex Editor, Inspector, Keyboard Mapper, Profiler and Theme Editor. Each of the apps by themselves are worthy of their own self-contained tutorial which we hope to soon see materialize. These apps are all native to SerenityOS as they are for SerenityOS, by SerenityOS and of SerenityOS. Python was included simply because this is the `Development` section of Serenity menu. Python is part of the ` Ports ` package and is actively being managed and maintained by [@linusg](https://github.com/linusg).
```
Comment (EWouters) [inline Documentation/FontEditor/UsingFontEditor.md:91]: I'm not sure this is very relevant to FontEditor

File: Documentation/FontEditor/UsingFontEditor.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+### Save font
+![](FontEditor_Save_font_as.png)
+
+Save your font by following the recommended ` FontName + FontStyle + FontPresentationSize + .font ` naming convention.
+
+### Continue Editing
+![](FontEditor_Continue_editing_current_font.png)
+
+The figure above shows some glyphs already created. Those with sharp eyes can immediately see that the glyphs for M and W, O and Q came from the same base. W was copied over from M and was flipped horizontally. Q was copied over from O and descender was added to achieve the final glyph. Continue editing your font by adding more glyphs into it. Take your time in creating the font, only you know the reason why the glyph is formed the way it is. You know why the height and the width is so, why the curvature is just so. Aside from achieving balance, maintain the spirit of fun in making your font. The more glyphs are added, the easier it is to get the general feel and character of the font. Remember to always save and to save often.
```
Comment (EWouters) [inline Documentation/FontEditor/UsingFontEditor.md:106]: descender -> a descender

File: Documentation/FontEditor/UsingFontEditor.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+
+### fonts.serenityos.net
+![](Fonts_SerenityOS_dot_Net.png)
+
+It is highly recommended to use our own [fonts portal](https://fonts.serenityos.net) as one of your primary resource for glyph and font information. The others are [Unicode charts [unicode.org]](https://www.unicode.org/charts/) and our [wiki](https://wiki.serenityos.net).
+
+### Search glyph
+![](Fonts_SerenityOS_dot_Net_search_result.png)
+
+Figure above depicts a search session on [fonts.serenityos.net](https://fonts.serenityos.net) portal.
```
Comment (EWouters) [inline Documentation/FontEditor/UsingFontEditor.md:116]: The figure above depicts a search session on the fonts.serenityos.net portal.

**Review by EWouters — CHANGES_REQUESTED** [review]
I learned a lot about FontEditor reading this, very insightful. I also spotted a few things I would word a bit differently.

File: Documentation/FontEditor/UsingFontEditor.md (hunk 1 above)
Comment (djwisdom) [reply Documentation/FontEditor/UsingFontEditor.md:14]: Wow first comment.  :^), thanks much for the input.

Comment (djwisdom) [reply Documentation/FontEditor/UsingFontEditor.md:14]: Updated as such.

File: Documentation/FontEditor/UsingFontEditor.md (hunk 2 above)
Comment (djwisdom) [reply Documentation/FontEditor/UsingFontEditor.md:52]: Updated as such.

File: Documentation/FontEditor/UsingFontEditor.md (hunk 3 above)
Comment (djwisdom) [reply Documentation/FontEditor/UsingFontEditor.md:57]: Updated as such.

File: Documentation/FontEditor/UsingFontEditor.md (hunk 4 above)
Comment (djwisdom) [reply Documentation/FontEditor/UsingFontEditor.md:60]: Updated as such.

File: Documentation/FontEditor/UsingFontEditor.md (hunk 6 above)
Comment (djwisdom) [reply Documentation/FontEditor/UsingFontEditor.md:76]: Updated as such.

File: Documentation/FontEditor/UsingFontEditor.md (hunk 8 above)
Comment (djwisdom) [reply Documentation/FontEditor/UsingFontEditor.md:91]: I just let the first sentence remain. "Figure above shows where you can find FontEditor from inside the SerenityOS desktop.".

File: Documentation/FontEditor/UsingFontEditor.md (hunk 9 above)
Comment (djwisdom) [reply Documentation/FontEditor/UsingFontEditor.md:106]: Updated as such.

File: Documentation/FontEditor/UsingFontEditor.md (hunk 10 above)
Comment (djwisdom) [reply Documentation/FontEditor/UsingFontEditor.md:116]: Updated as such.

**Comment by djwisdom** [comment]
Re-uploading document along with the recommended rewordings. Immense thank you to @EWouters for the much needed inputs :^)

File: Documentation/FontEditor/UsingFontEditor.md (hunk 7 above)
Comment (djwisdom) [reply Documentation/FontEditor/UsingFontEditor.md:81]: Updated as such.

File: Documentation/FontEditor/UsingFontEditor.md (hunk 5 above)
Comment (djwisdom) [reply Documentation/FontEditor/UsingFontEditor.md:66]: Updated as such.

**Pull request merged**